### PR TITLE
export and consume the properties in testenv.property file

### PR DESCRIPTION
**Pull request by xius666**
issue #2963
using the shell script to export the correct variable instead of changing the build.xml
a cleaner approach of pr #3069

### CODE REVIEW

**Comment by xius666** [comment]
grinder:https://ci.adoptopenjdk.net/job/Grinder/1963/

File: testenv/testenv.properties [hunk 1]
```diff
@@ -12,7 +12,7 @@ JDK8_REPO=https://github.com/adoptium/jdk8u.git
 JDK8_BRANCH=master
 JDK11_REPO=https://github.com/adoptium/jdk11u.git
 JDK11_BRANCH=master
-JDK17_REPO=https://github.com/adoptium/jdk17.git
+JDK17_REPO=https://github.com/adoptium/jdk17.git
```
Comment (llxia) [inline testenv/testenv.properties:15]: Extra space is added to this file.

**Comment by llxia** [comment]
https://ci.adoptopenjdk.net/job/Grinder/2171/
https://ci.adoptopenjdk.net/job/Grinder/2172/

**Comment by llxia** [comment]
```
01:29:58.246  Set values based on ./testenv/testenv.properties:
01:29:58.246  TKG_REPO=https://github.com/adoptium/TKG.git
01:29:58.246  TKG_BRANCH=master
01:29:58.246  OPENJ9_REPO=https://github.com/eclipse-openj9/openj9.git
01:29:58.246  OPENJ9_BRANCH=master
01:29:58.246  STF_REPO=https://github.com/adoptium/STF.git
01:29:58.246  STF_BRANCH=master
01:29:58.246  OPENJ9_SYSTEMTEST_REPO=https://github.com/eclipse-openj9/openj9-systemtest.git
01:29:58.246  OPENJ9_SYSTEMTEST_BRANCH=master
01:29:58.246  ADOPTOPENJDK_SYSTEMTEST_REPO=https://github.com/adoptium/aqa-systemtest.git
01:29:58.246  ADOPTOPENJDK_SYSTEMTEST_BRANCH=master
01:29:58.246  JDK8_REPO=https://github.com/adoptium/jdk8u.git
01:29:58.246  JDK8_BRANCH=master
01:29:58.246  JDK11_REPO=https://github.com/adoptium/jdk11u.git
01:29:58.246  JDK11_BRANCH=master
01:29:58.246  JDK17_REPO=https://github.com/adoptium/jdk17.git
01:29:58.246  JDK17_BRANCH=master
01:29:58.246  JDK8_OPENJ9_REPO=https://github.com/ibmruntimes/openj9-openjdk-jdk8.git
01:29:58.246  JDK8_OPENJ9_BRANCH=master
01:29:58.246  JDK11_OPENJ9_REPO=https://github.com/ibmruntimes/openj9-openjdk-jdk11.git
01:29:58.246  JDK11_OPENJ9_BRANCH=master
01:29:58.246  JDK17_OPENJ9_REPO=https://github.com/ibmruntimes/openj9-openjdk-jdk17.git
01:29:58.246  JDK17_OPENJ9_BRANCH=masterJDK_REPO=https://github.com/adoptium/jdk8u.git
01:29:58.246  JDK_BRANCH=master
```
https://ci.adoptopenjdk.net/job/Grinder/2171/console

@xius666 one minor thing, JDK17_OPENJ9_BRANCH and JDK_REPO are in one line. Could you add the next line in between? Thanks

**Comment by xius666** [comment]
https://ci.adoptopenjdk.net/job/Grinder/2305/console

**Review by llxia — APPROVED** [review]
Thanks @xius666

**Comment by llxia** [comment]
AIX: 
https://ci.adoptopenjdk.net/job/Grinder/2310
https://ci.adoptopenjdk.net/job/Grinder/2311

File: compile.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+if [ $(uname) = AIX ] || [ $(uname) = SunOS ] || [ $(uname) = *BSD ]; then
+    MAKE=gmake
+else
+    MAKE=make
+fi
+if [ $USE_TESTENV_PROPERTIES == true ]; then
+    while read line; do
+        export $line
+    done <./testenv/testenv.properties
+    repo_j9=JDK${JDK_VERSION}_OPENJ9_REPO
```
Comment (sophia-guo) [inline compile.sh:15]: can we move line 15 - 22 to the following if [ $JDK_IMPL == "openj9" ] block or else block depends on properties?

something like:

```
if [ $JDK_IMPL == "openj9" ]; then
     repo=JDK${JDK_VERSION}_OPENJ9_REPO
     branch=JDK${JDK_VERSION}_OPENJ9_BRANCH
else
     repo=JDK${JDK_VERSION}_REPO
     branch=JDK${JDK_VERSION}_BRANCH
fi

    eval repo2='$'$repo
    eval branch2='$'$branch

    export JDK_REPO=$repo2
    export JDK_BRANCH=$branch2
```

**Comment by xius666** [comment]
https://ci.adoptopenjdk.net/job/Grinder/2316/

**Review by sophia-guo — APPROVED** [review]
LGTM.

File: compile.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    MAKE=gmake
+else
+    MAKE=make
+fi
+if [ $USE_TESTENV_PROPERTIES == true ]; then
+    while read line; do
+        export $line
+    done <./testenv/testenv.properties
+    if [ $JDK_IMPL == "openj9" ]; then
+        repo=JDK${JDK_VERSION}_OPENJ9_REPO
```
Comment (sophia-guo) [inline compile.sh:16]: Probably rename repo, branch to repoTemp and branchTemp. And then rename repo2 and branch2 to repo and branch. It's just my personal preference, which feels like more clear by naming.

**Review by llxia — APPROVED** [review]
LGTM

**Comment by llxia** [comment]
https://ci.adoptopenjdk.net/job/Grinder/2323/console

**Pull request merged**